src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -0,0 +1,51 @@
+namespace System
+{
+    internal static class HexConverter
+    {
+        public static char ToCharUpper(int value)
+        {
+            value &= 0xf;
+            return (char)(value > 9 ? value - 10 + 'A' : value + '0');
+        }
+
+        public static char ToCharLower(int value)
+        {
+            value &= 0xf;
+            return (char)(value > 9 ? value - 10 + 'a' : value + '0');
+        }
+
+#if NETCOREAPP || NETSTANDARD2_1
+        public static unsafe string ToStringLower(ReadOnlySpan<byte> bytes)
+        {
+            fixed (byte* bytesPtr = bytes)
```

Comment:
```suggestion
            Debug.Assert(!bytes.IsEmpty);
            fixed (byte* bytesPtr = &MemoryMarshal.GetReference(bytes))
```
Should be a little bit faster. But I don't know if it's worth in contrast to the delegate invocation in `string.Create`.